samples/19.b.single-sign-on-for-intranet-apps/README.md

```diff
@@ -0,0 +1,199 @@
+
+# Test out the hosted sample
+
+You can browse to https://webchat-sample-sso-intranet.azurewebsites.net/ to try out this demo.
+
+# How to run locally
+
+This demo integrates with Azure Active Directory. You will need to set it up in order to host the demo.
+
+1. [Clone the code](#clone-the-code)
+1. [Setup OAuth via Azure Active Directory](#setup-oauth-via-azure-active-directory)
+1. [Setup Azure Bot Services](#setup-azure-bot-services)
+1. [Prepare and run the code](#prepare-and-run-the-code)
+
+## Clone the code
+
+To host this demo, you will need to clone the code and run locally.
+
+1. Clone this repository
+1. Create two files for environment variables, `/bot/.env` and `/web/.env`
+   -  In `/web/.env`:
+      -  Write `OAUTH_REDIRECT_URI=http://localhost:3000/api/oauth/callback`
+         -  When Azure Active Directory completes the authorization flow, it will send the browser to this URL. This URL must be accessible by the browser from the end-user machine
+
+## Setup OAuth via Azure Active Directory
+
+If you want to authenticate on Azure Active Directory, follow the steps below.
+
+-  Go to your [Azure Active Directory](https://ms.portal.azure.com/#blade/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/Overview)
+-  Create a new application
+   1. Select "App registrations"
+   1. Click "New registration"
+   1. Fill out "Name", for example, "Web Chat SSO Sample"
+   1. In "Redirect URI (optional)" section, add a new entry
+      1. Select "Public client (mobile & desktop)" as type
+         -  Instead of client secret, we are using PKCE ([RFC 7636](https://tools.ietf.org/html/rfc7636)) to exchange for authorization token, thus, we need to set it to ["Public client" instead of "Web"](https://docs.microsoft.com/en-us/azure/active-directory/develop/v1-protocols-oauth-code#use-the-authorization-code-to-request-an-access-token)
+      1. Enter `http://localhost:3000/api/oauth/callback` as the redirect URI
+         -  This must match `OAUTH_REDIRECT_URI` in `/web/.env` we saved earlier
+   -  Click "Register"
+-  Save the client ID
+   1. Select "Overview"
+   1. On the main pane, copy the content of "Application (client) ID" to `/web/.env`, it should looks be a GUID
+      -  `OAUTH_CLIENT_ID=12345678abcd-1234-5678-abcd-12345678abcd`
+
+## Setup Azure Bot Services
+
+> We prefer to use [Bot Channel Registration](https://ms.portal.azure.com/#create/Microsoft.BotServiceConnectivityGalleryPackage) during development. This will help you diagnose problems locally without deploying to the server and speed up development.
```

Comment:
```suggestion
> We prefer using [Bot Channel Registration](https://ms.portal.azure.com/#create/Microsoft.BotServiceConnectivityGalleryPackage) during development. This will help you diagnose problems locally without deploying to the server and speed up development.
```